src/full/Agda/TypeChecking/Records.hs

```diff
@@ -842,6 +847,7 @@ isSingletonRecord'
                      --   otherwise we would construct an infinite inhabitant (in an infinite time...).
   -> m (Maybe Term)  -- ^ The unique inhabitant, if any.  May contain dummy terms in irrelevant positions.
 isSingletonRecord' regardIrrelevance r ps rs = do
+  ifM (optEtaSingleton <$> pragmaOptions) (return Nothing) $ do
```

Comment:
This looks like a use-site logic (while the PR description talks about definitions-site).  
(NB: I think you missed a negation here.)

Reply:
You're right - it's currently a weird mix of definition-site and use-site. I'll change it to be use-site only, as that seems to be the more useful/backwards compatible version of this flag. And indeed, seems I missed a negation :sweat_smile: